src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/RuntimeBinder.cs

```diff
@@ -635,15 +635,17 @@ private Expr CreateArgumentEXPR(ArgumentObject argument, LocalVariableSymbol loc
                 callingType = (AggregateType)callingObjectType;
             }
 
-            List<CType> callingTypes = new List<CType>();
-
             // The C# binder expects that only the base virtual method is inserted
             // into the list of candidates, and only the type containing the base
             // virtual method is inserted into the list of types. However, since we
             // don't want to do all the logic, we're just going to insert every type
             // that has a member of the given name, and allow the C# binder to filter
             // out all overrides.
-            //
+            
+            // CONSIDER: using a hashset to filter out duplicate interface types.
+            // Adopt a smarter algorithm to filter types before creating the exception.
+            HashSet<CType> callingTypes = new HashSet<CType>();
```

Comment:
Upon further examination, I don't see a lot of tests for this area in particular. I'll preserve the existing order until we investigate further.